### PR TITLE
folder(*): adding SVM model to linearly classify a toy dataset from scratch

### PR DESCRIPTION
**Pull request by subham103**
@AnirudhDagar  please review!!

### CODE REVIEW

File: assignment_3/SVM_by_subham/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+## Approach
+
+1. Generated the toy dataset using make_blobs
+	- Approach
```
Comment (AnirudhDagar) [inline assignment_3/SVM_by_subham/README.md]: Get rid of this line "-Approach". Is it incomplete? looks abrupt.

File: assignment_3/SVM_by_subham/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+2. Converted the data into a dictionary data_dict with positive class as 1 and negative as -1.
+
+3. Made a class named SVM with 3 method.
+    -fit
```
Comment (AnirudhDagar) [inline assignment_3/SVM_by_subham/README.md]: these points are not getting rendered as expected in a list in markdown.
also when making a list in markdown you need to leave a blank line.
use a space between '-' and 'fit' like below
 ```suggestion
 
    - fit
    - predict 
    - visualize 
```

File: assignment_3/SVM_by_subham/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    -predict
```
Comment (AnirudhDagar) [inline assignment_3/SVM_by_subham/README.md]:
```suggestion
```

File: assignment_3/SVM_by_subham/README.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    -visualize
```
Comment (AnirudhDagar) [inline assignment_3/SVM_by_subham/README.md]:
```suggestion

```

File: assignment_3/SVM_by_subham/README.md (hunk 1 above)
Comment (AnirudhDagar) [reply assignment_3/SVM_by_subham/README.md]: These are minute changes but please take care about them.

File: assignment_3/SVM_by_subham/README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+4. In method fit, searched for the optimum w and optimum b in the range of -(maximum feature value) to +(maximum feature value) initialy and then decreased the range with 0.1 then with 0.01 and lastly with 0.001. If any w and b are the required w and b then checked the condition yi*(xi*w+b) >= 1 to be true.
+
+5. Visualised the data with the method visualize. 
+
+## Steps to run
+
+1. If want to run with the default dataset:
+    -clone the repo.
```
Comment (AnirudhDagar) [inline assignment_3/SVM_by_subham/README.md]: same thing here with markdown rendering.

File: assignment_3/SVM_by_subham/README.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    -run the svmfrombasic.py script in terminal by the command      => python svmfrombasic.py
+
+2. If want to use a different dataset:
+    -clone the repo
```
Comment (AnirudhDagar) [inline assignment_3/SVM_by_subham/README.md]: same

File: assignment_3/SVM_by_subham/README.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+2. Converted the data into a dictionary data_dict with positive class as 1 and negative as -1.
+
+3. Made a class named SVM with 3 method.
+    -fit
+    -predict
+    -visualize
+
+4. In method fit, searched for the optimum w and optimum b in the range of -(maximum feature value) to +(maximum feature value) initialy and then decreased the range with 0.1 then with 0.01 and lastly with 0.001. If any w and b are the required w and b then checked the condition yi*(xi*w+b) >= 1 to be true.
+
+5. Visualised the data with the method visualize.
```
Comment (AnirudhDagar) [inline assignment_3/SVM_by_subham/README.md]: Here you can add the visualized result .png or .jpeg file of the matplotlib figure.
like this 
image

File: assignment_3/SVM_by_subham/svmfrombasic.py [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+            
+            norms = sorted([mod_w for mod_w in opt_dict])
+            opt_choice = opt_dict[norms[0]]
+            self.w = opt_choice[0]
+            self.b = opt_choice[1]
+            latest_optimum = opt_choice[0][0]+step*2
+ 
+    
+    #Method for prediction of new data points
+    def predict(self, features):
```
Comment (AnirudhDagar) [inline assignment_3/SVM_by_subham/svmfrombasic.py:93]: you may have called predict somewhere to show its use. But its okay, just a comment. Keep it as it is for now.

**Review by AnirudhDagar — CHANGES_REQUESTED** [review]
Great visualization. 👍 for that. Appreciate your efforts.
Just make the required changes in the readme, I'll merge.
Also, the problem statement asked to compare the performance in terms of time to fit the model. It's fine for now though.

File: assignment_3/SVM_by_subham/README.md (hunk 1 above)
Comment (subham103) [reply assignment_3/SVM_by_subham/README.md]: Actually!! I forgot to delete that line. My bad..

**Comment by subham103** [comment]
@AnirudhDagar  I have made the required changes. Please review again..

**Review by AnirudhDagar — APPROVED** [review]

**Pull request merged**